public/scripts/chats.js

```diff
@@ -189,55 +189,66 @@ export async function populateFileAttachment(message, inputId = 'file_form_input
         if (!message.extra) message.extra = {};
         const fileInput = document.getElementById(inputId);
         if (!(fileInput instanceof HTMLInputElement)) return;
-        const file = fileInput.files[0];
-        if (!file) return;
-
-        const slug = getStringHash(file.name);
-        const fileNamePrefix = `${Date.now()}_${slug}`;
-        const fileBase64 = await getBase64Async(file);
-        let base64Data = fileBase64.split(',')[1];
-
-        // If file is image
-        if (file.type.startsWith('image/')) {
-            const extension = file.type.split('/')[1];
-            const imageUrl = await saveBase64AsFile(base64Data, name2, fileNamePrefix, extension);
-            message.extra.image = imageUrl;
-            message.extra.inline_image = true;
-        } else {
-            const uniqueFileName = `${fileNamePrefix}.txt`;
-
-            if (isConvertible(file.type)) {
-                try {
-                    const converter = getConverter(file.type);
-                    const fileText = await converter(file);
-                    base64Data = window.btoa(unescape(encodeURIComponent(fileText)));
-                } catch (error) {
-                    toastr.error(String(error), t`Could not convert file`);
-                    console.error('Could not convert file', error);
+        const files = fileInput.files;
+        if (!files || files.length === 0) return;
+
+        // Initialize image array if we're handling images
+        message.extra.image = [];
```

Comment:
What if you're attaching files to existing messages that already contain a single image?

Reply:
As of right now if you attach to an already existing message, any previous attachments get overwritten.

Reply:
Not a very good UX in my opinion. I think we should make it so that it appends to an already existing lists of files or images.

Reply:
Hmmm, I think the fix is easy just check if message.extra.image already exists and only initialize it if it doesn't.
```suggestion
        if(!message.extra.image) message.extra.image = [];
```

I will test later when I have time. Maybe it would be also good to check that it's an array?

Reply:
> Maybe it would be also good to check that it's an array?

Yeah. If image is an array - skip. If truthy but not an array - convert it to an array.

Reply:
Okay I think that works now